diff --color -uNr dqlite-1.18.0/src/raft/heap.c dqlite-1.18.0.mod/src/raft/heap.c
--- dqlite-1.18.0/src/raft/heap.c	2024-10-23 11:25:04.000000000 +0800
+++ dqlite-1.18.0.mod/src/raft/heap.c	2024-10-24 21:57:58.409370853 +0800
@@ -28,6 +28,15 @@
 	return realloc(ptr, size);
 }
 
+void *aligned_alloc(size_t alignment, size_t size) {
+    void *ptr = NULL;
+    if (posix_memalign(&ptr, alignment, size) != 0) {
+        return NULL; // Allocation failed
+    }
+    return ptr;
+}
+
+
 static void *defaultAlignedAlloc(void *data, size_t alignment, size_t size)
 {
 	(void)data;
