diff --color -uNr lxd-6.1/lxd/lxc/main.go lxd-6.1.mod/lxd/lxc/main.go
--- lxd-6.1/lxd/lxc/main.go	2024-07-08 15:45:26.000000000 +0800
+++ lxd-6.1.mod/lxd/lxc/main.go	2024-10-31 00:28:00.055077172 +0800
@@ -289,14 +289,14 @@
 	err = app.Execute()
 	if err != nil {
 		// Handle non-Linux systems
-		if err == config.ErrNotLinux {
-			fmt.Fprintf(os.Stderr, i18n.G(`This client hasn't been configured to use a remote LXD server yet.
-As your platform can't run native Linux instances, you must connect to a remote LXD server.
+//		if err == config.ErrNotLinux {
+//			fmt.Fprintf(os.Stderr, i18n.G(`This client hasn't been configured to use a remote LXD server yet.
+//As your platform can't run native Linux instances, you must connect to a remote LXD server.
 
-If you already added a remote server, make it the default with "lxc remote switch NAME".
-To easily setup a local LXD server in a virtual machine, consider using: https://multipass.run`)+"\n")
-			os.Exit(1)
-		}
+//If you already added a remote server, make it the default with "lxc remote switch NAME".
+//To easily setup a local LXD server in a virtual machine, consider using: https://multipass.run`)+"\n")
+//			os.Exit(1)
+//		}
 
 		// Default error handling
 		fmt.Fprintf(os.Stderr, "Error: %v\n", err)
